website/docs/migration.md

```diff
@@ -0,0 +1,244 @@
+   test>
+   ```
+
+### Manual and automated testing with `diff-proxy` mode
+
+`diff-proxy` mode will forward client requests to a MongoDB-compatible database, but return the proxy responses to the client, and will log the difference between them.
+
+Continuing with the same example above, we can further examine the diff output.
+
+1. Run FerretDB in `diff-proxy` mode.
+
+   ```bash
+   me@foobar:~/FerretDB$ bin/task run-proxy
+   ```
+
+2. Re-run the query.
+
+   ```js
+   test> db.posts.aggregate(
+   ...   [
+   ...     { $sort: { date: 1, author: 1 } },
+   ...     {
+   ...       $group:
+   ...         {
+   ...           _id: "$author",
+   ...           firstPost: { $first: "$date" }
+   ...         }
+   ...     }
+   ...   ]
+   ... );
+   [
+     { _id: 'Alice', firstPost: ISODate("2023-08-20T10:33:23.134Z") },
+     { _id: 'Bob', firstPost: ISODate("2023-08-28T10:33:23.134Z") }
+   ]
+   test>
+   ```
+
+In the diff output below we have discovered that the query cannot be serviced by our application because the `$first` accumulator operator is not implemented in FerretDB.
+
+```bash
+2023-08-29T13:25:09.048+0200  WARN  // 127.0.0.1:33522 -> 127.0.0.1:27017  clientconn/conn.go:360 Header diff:
+--- res header
++++ proxy header
+@@ -1 +1 @@
+-length:   140, id:    2, response_to:  156, opcode: OP_MSG
++length:   181, id:  360, response_to:  156, opcode: OP_MSG
+
+Body diff:
+--- res body
++++ proxy body
+@@ -7,13 +7,41 @@
+         "$k": [
+-          "ok",
+-          "errmsg",
+-          "code",
+-          "codeName"
++          "cursor",
++          "ok"
+         ],
++        "cursor": {
++          "$k": [
++            "firstBatch",
++            "id",
++            "ns"
++          ],
++          "firstBatch": [
++            {
++              "$k": [
++                "_id",
++                "firstPost"
++              ],
++              "_id": "Alice",
++              "firstPost": {
++                "$d": 1692527603134
++              }
++            },
++            {
++              "$k": [
++                "_id",
++                "firstPost"
++              ],
++              "_id": "Bob",
++              "firstPost": {
++                "$d": 1693218803134
++              }
++            }
++          ],
++          "id": {
++            "$l": "0"
++          },
++          "ns": "test.posts"
++        },
+         "ok": {
+-          "$f": 0
+-        },
+-        "errmsg": "$group accumulator \"$first\" is not implemented yet",
+-        "code": 238,
+-        "codeName": "NotImplemented"
++          "$f": 1
++        }
+       },
+```
+
+### Response metrics
+
+Metrics are captured and written to standard output (`stdout`) upon exiting in [Debug builds](https://pkg.go.dev/github.com/FerretDB/FerretDB@v1.8.0/build/version#hdr-Debug_builds).
+This is a useful way to quickly determine what commands are not implemented for the client requests sent by your application.
+
+```bash
```

Comment:
That not bash

Reply:
The linter needs an language identifier in a code block I believe.

Reply:
kept as `sh` for now